lib/cpp/CMakeLists.txt

```diff
@@ -178,6 +182,84 @@ install(DIRECTORY "src/thrift" DESTINATION "${INCLUDE_INSTALL_DIR}"
 install(DIRECTORY "${CMAKE_BINARY_DIR}/thrift" DESTINATION "${INCLUDE_INSTALL_DIR}"
     FILES_MATCHING PATTERN "*.h")
 
+# Introduce variables:
+# * CMAKE_INSTALL_LIBDIR
+# * CMAKE_INSTALL_BINDIR
+# * CMAKE_INSTALL_INCLUDEDIR
+include(GNUInstallDirs)
+
+# Layout. This works for all platforms:
+#   * <prefix>/lib*/cmake/<PROJECT-NAME>
+#   * <prefix>/lib*/
+#   * <prefix>/include/
+set(config_install_dir "${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME}")
+
+set(generated_dir "${CMAKE_CURRENT_BINARY_DIR}/generated")
+
+# Configuration
+set(version_config "${generated_dir}/${PROJECT_NAME}ConfigVersion.cmake")
+set(project_config "${generated_dir}/${PROJECT_NAME}Config.cmake")
+set(TARGETS_EXPORT_NAME "${PROJECT_NAME}Targets")
+set(namespace "${PROJECT_NAME}::")
+
+# Include module with fuction 'write_basic_package_version_file'
+include(CMakePackageConfigHelpers)
+
+# Configure '<PROJECT-NAME>ConfigVersion.cmake'
+# Use:
+#   * PROJECT_VERSION
+write_basic_package_version_file(
+    "${version_config}" COMPATIBILITY SameMajorVersion
```

Comment:
When we go from 0.16.0 to 1.0 this means we won't be compatible?
Generally speaking, we potentially make breaking changes in the public interface in any release.
We certainly have between 0.12.0 and 0.13.0.

Reply:
I put SameMajorVersion only because it's often used, but if I understand correctly you'd prefer ExactVersion? The choice is currently AnyNewerVersion|SameMajorVersion|SameMinorVersion|ExactVersion, see https://cmake.org/cmake/help/v3.0/module/CMakePackageConfigHelpers.html

Reply:
SameMinorVersion seems reasonable for us.  Basically we would expect 0.12.0 and 0.12.1 to be binary compatible in terms of public interfaces in a language runtime, but you wouldn't necessarily be able to use a 0.13.0 runtime for C++ with something linked against 0.12.0.

Reply:
Done.